web/modules/custom/collection/src/Form/JoinCollectionForm.php

```diff
@@ -128,6 +129,10 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       ->setState(OgMembershipInterface::STATE_ACTIVE)
       ->save();
 
+    // @todo: when og gets its own caching system, use its context instead.
+    // Invalidate cache so that the add content block can be rebuilt.
+    Cache::invalidateTags(['block_view']);
```

Comment:
If you have a @todo there should always be a link to a ticket. Otherwise this is not tracked and then it will never be solved.

This workaround is wrong for several reasons:
1. This is a Form class, it is not concerned with where it is being displayed. You should not have any code here relating to blocks. Having this here makes the form no longer reusable.
2. The 'block_view' tag sounds very generic to me. Are you clearing *all* the blocks with this? Don't do that! :)
3. This needs more research. I don't see why OG would not have the necessary cache contexts, after all it is using generic entities.